certifier/README.md

```diff
@@ -0,0 +1,65 @@
+# POST certifier service
+A certifier service that creates certificates for a node confirming it holds a valid POST proof.
+
+The client (presumably the spacemesh node) submits a POST proof with its metadata to the certifier on /certify HTTP endpoint. The certifier validates the proof and, if valid - signs the nodeID and returns the signature. If the proof is invalid it returns a 403 status code.
+
+The client can later use this certificate to register in a poet. The poet is supposed to know the certifier's public key and verify the signature over a node ID.
+
+## Usage
+```
+Usage: certifier [OPTIONS] [COMMAND]
+
+Commands:
+  generate-keys  generate keypair and write it to standard out. the keypair is encoded as json
+  help           Print this message or the help of the given subcommand(s)
+
+Options:
+  -c, --config <CONFIG>  [env: CERTIFIER_CONFIG_PATH=] [default: config.yml]
+  -h, --help             Print help
+  -V, --version          Print version
+```
+
+### running a service
+To run a service, execute the certifier w/o additional command and provide a path to config (either via --config or CERTIFIER_CONFIG_PATH env variable).
+
+#### Configuration
+The config structure is defined [here](src/configuration.rs). An example config:
+
+```yaml
+listen: "127.0.0.1:8080"
+signing_key: <BASE64-encoded ed25519 private key>
+post_cfg:
+  k1: 26
+  k2: 37
+  k3: 37
+  pow_difficulty: "000dfb23b0979b4b000000000000000000000000000000000000000000000000"
+
+init_cfg:
+  min_num_units: 4
+  max_num_units: 99999
+  labels_per_unit: 4294967296
+  scrypt:
+    n: 8192
+    r: 1
+    p: 1
+
+metrics: "127.0.0.1:9090"
```

Comment:
Can this somehow be gamed by only init the minimum amount generating a proof of that to get a certificate and then keep initializing, i.e. the effort to get a certificate isn't dependent on the post size?

Would this lead to reasonable smeshers to always init the minimum for the certificate (unless they already have a proof for a bigger size)?

Reply:
It probably is OK though since the PoW is also the same difficulty independent of PoST size.

Reply:
Yes, one could initialize 4 units, create a POST proof, certify with it, and then keep initializing. It doesn't matter because we only want to confirm that a smesher spent significant resources and has a valid POST (at some point in time).